library/std/src/sys/thread/unix.rs

```diff
@@ -52,10 +52,13 @@ impl Thread {
         name: Option<&str>,
         f: Box<dyn FnOnce()>,
     ) -> io::Result<Thread> {
-        let data = Box::into_raw(Box::new(ThreadData { name: name.map(Box::from), f }));
-        let mut native: libc::pthread_t = mem::zeroed();
+        let data = Box::new(ThreadData { name: name.map(Box::from), f });
```

Comment:
Seems like this declaration could be moved down in its entirety. I'm not sure why you only did that for the `Box::into_raw` part.